src/pages/programsWip/form/SetupFormContents.tsx

```diff
@@ -1,14 +1,156 @@
+import { useTimeZoneConversion } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { InputFieldFF, SingleSelectFieldFF } from '@dhis2/ui'
 import React from 'react'
-import { SectionedFormSection } from '../../../components'
+import { useField, useFormState } from 'react-final-form'
+import { useHref } from 'react-router'
+import {
+    DefaultIdentifiableFields,
+    DescriptionField,
+    EditableFieldWrapper,
+    SectionedFormSection,
+    StandardFormField,
+} from '../../../components'
+import { defaultDateTimeFormatter } from '../../../components/date'
+import {
+    ModelSingleSelectFormField,
+    useRefreshModelSingleSelect,
+} from '../../../components/metadataFormControls/ModelSingleSelect'
+import { DEFAULT_CATEGORYCOMBO_SELECT_OPTION } from '../../../lib'
+import { DisplayableModel } from '../../../types/models'
+import classes from '../../dataElements/fields/CategoryComboField.module.css'
+
+const featureTypes = [
+    {
+        value: '',
+        label: i18n.t('<No value>'),
+    },
+    {
+        value: 'NONE',
+        label: i18n.t('None'),
+    },
+    {
+        value: 'POINT',
+        label: i18n.t('Point'),
+    },
+    {
+        value: 'POLYGON',
+        label: i18n.t('Polygon'),
+    },
+]
+const CATEGORY_COMBOS_QUERY = {
+    resource: 'categoryCombos',
+    params: {
+        filter: ['dataDimensionType:eq:ATTRIBUTE'],
+        fields: ['id', 'displayName', 'name'],
+    },
+}
+
+const addDefaultCategoryComboTransform = <TCatCombo extends DisplayableModel>(
+    catCombos: TCatCombo[]
+) => [DEFAULT_CATEGORYCOMBO_SELECT_OPTION, ...catCombos]
 
 export const SetupFormContents = React.memo(function SetupFormContents({
     name,
 }: {
     name: string
 }) {
+    const { input: versionInput, meta: versionMeta } = useField('version', {
+        type: 'number',
+        format: (value) => value?.toString(),
+        validate: (value) =>
```

Comment:
maybe would be good to add a custom validation message in the property definition in zod schema, e.g.
```version: z.coerce.number().int(i18n.t('Only integers are allowed')).optional()` so we don't get the standard `Expected integer, received float` when entering a decimal value?